src/StreamingBodyParser/ParserInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace React\Http\StreamingBodyParser;
+
+use Evenement\EventEmitterInterface;
+use React\Http\Request;
+
+/**
+ * Parser can emit the following events:
+ *     end - When done or canceled (required)
+ *     body - Raw request body (optional)
+ *     post - Post field (optional)
+ *     file - Uploaded file (optional)
+ */
+interface ParserInterface extends EventEmitterInterface
+{
+    /**
+     * Factory method creating the parser.
+     *
+     * @param Request $request
+     * @return ParserInterface
+     */
+    public static function create(Request $request);
```

Comment:
Since the parser implementations are now `@internal`, i think this method can be removed from the interface and the implementations and simple constructor based instantiantion can be used in `Factory`.